Lib/pathlib.py

```diff
@@ -878,7 +899,15 @@ def is_junction(self):
         """
         Whether this path is a junction.
         """
-        return os.path.isjunction(self)
+        import stat
+        try:
+            return self.lstat().st_reparse_tag == stat.IO_REPARSE_TAG_MOUNT_POINT
```

Comment:
> Minor: Is there a reason we can't import `IO_REPARSE_TAG_MOUNT_POINT` into the module scope along with the other stat names?

It's only available on Windows

> Also: is this a known idiom on Unix systems? The documentation for [`st_reparse_tag`](https://docs.python.org/3.13/library/os.html#os.stat_result.st_reparse_tag) and [`IO_REPARSE_TAG_*`](https://docs.python.org/3.13/library/stat.html#stat.IO_REPARSE_TAG_SYMLINK) both fall under sections marked as Windows-only. If these are common Unix idioms, we should update the documenatation for `os` and `stat`.

No; it's a purely Windows idiom. On other platforms, the `.st_reparse_tag` lookup raises an AttributeError, which is caught below and the method returns `False`.

Reply:
Junctions seem to be a Windows-only feature, whereas other types of file (directories, symlinks, mountpoints) have analogues in some virtual filesystems. And so it's not clear to me what `_PathBase.is_junction()` should do, really. Immediately `return False`? Raise `UnsupportedOperation`? The current code effectively always returns false, unless the user returns from `stat()` an object with a `st_reparse_tag` attribute set correctly, and is running on Windows.

Reply:
I've revised this code - `is_junction()` now ignores the result of `lstat()` and returns false. I _think_ this is a good way forward, but I'm not sure.

Reply:
I think this might be overcomplicating it -- for the default implementation, perhaps just return ``False``, as junctions are clearly fairly domain-specific. The current method has a dollop of code just to say "no" or to raise an error!

Reply:
I think I agree, but just to be clear: that would make `is_junction()` the only `_PathBase` that does not raise `UnsupportedOperation` by default. A worthwhile compromise, given it's such an edge case?

Reply:
Thanks for your help with this @AA-Turner. I've made `is_junction()` immediately return false.